lib/build_cache.py

```diff
@@ -938,8 +982,9 @@ def pull_eager(self, img, src_ref, last_layer=None):
       if (dl_git_hash is not None):
          # Downloaded image is in cache, check it out.
          ch.INFO("pulled image: found in build cache")
-         self.checkout(img, dl_git_hash, None)
-         self.ready(img)
+         # Remove tag for previously deleted branch, if it exists.
+         self.git(["tag", "-d", "&%s" % img.ref.for_path], fail_ok=True)
```

Comment:
In general we want to avoid direct Git operations, especially ones like this that encode stuff in strings. We already have a delete method, yes? An undelete method is likely a good thing to add.